plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java

```diff
@@ -83,6 +84,7 @@ private ConnectorPageSink createPageSink(ConnectorSession session, IcebergWritab
         Schema schema = SchemaParser.fromJson(tableHandle.getSchemaAsJson());
         String partitionSpecJson = tableHandle.getPartitionsSpecsAsJson().get(tableHandle.getPartitionSpecId());
         PartitionSpec partitionSpec = PartitionSpecParser.fromJson(schema, partitionSpecJson);
+        verifyNotPartitionedByNestedField(schema, partitionSpec);
```

Comment:
Sink setup feels a little late in the query execution to be catching this. Can we have these checks in IcebergMetadata, [like this one](https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1711)?

Reply:
Removed the 1st commit.